data/binding/binditems.go

```diff
@@ -10,16 +10,22 @@ type Bool interface {
 	Set(bool)
 }
 
+// BoolPointer supports binding a bool value in a Fyne application
+type BoolPointer interface {
+	Bool
+	DataPointer
+}
+
 // NewBool returns a bindable bool value that is managed internally.
 func NewBool() Bool {
 	blank := false
 	return &boundBool{val: &blank}
 }
 
 // BindBool returns a new bindable value that controls the contents of the provided bool variable.
-func BindBool(v *bool) Bool {
+func BindBool(v *bool) BoolPointer {
```

Comment:
I wonder if this should be `NewBoolPointer` for consistency

Reply:
I don't think so personally. As you get into more complex bindings it get's really wordy - i.e. `NewStringListPointer`.
This is, after all, not returning a pointer it is returning a `Bool` binding with the additional method attached.
Revising so it is less inconsistent.

Reply:
How about `func NewBound<Data>(v *<data>) <Data>Pointer`?

Reply:
It should not have the `New` prefix as it is not creating a new. `NewBool` creates a new bool, `BindBool` binds to an existing one.